src/Symfony/Component/HttpClient/HttpClientTrait.php

```diff
@@ -313,21 +312,127 @@ private static function normalizeHeaders(array $headers): array
      *
      * @throws InvalidArgumentException When an invalid body is passed
      */
-    private static function normalizeBody($body)
+    private static function normalizeBody($body, array &$normalizedHeaders = [])
     {
         if (\is_array($body)) {
-            array_walk_recursive($body, $caster = static function (&$v) use (&$caster) {
-                if (\is_object($v)) {
+            static $cookie;
```

Comment:
why is this a static variable instead of a local variable ? Do we really need a hidden global state here ?

Reply:
We need this to generate a random boundary without consuming too much entropy from the system.
We don't care about the hidden state because this it just a random string that is updated after use.

Reply:
I'm not against using a variable. But it could be a normal one, so that each call to `prepareBody` is independant.

Reply:
Or you should not reuse that cookie when generating the multipart boundary that is visible to the outside.

Reply:
Each call is already independent from the previous one thanks to the randomization provided by the xxh128 hashing.
I don't want to call `random_bytes` all the time.